draft-ietf-quic-manageability.md

```diff
@@ -425,12 +426,11 @@ datagram can also carry 0-RTT data, as shown in {{fig-client-initial-0rtt}}.
 {: #fig-client-initial-0rtt
    title="Typical 0-RTT Client Initial datagram pattern"}
 
-In a 0-RTT Client Initial datagram, the PADDING frame is only present if
-necessary to increase the size of the datagram with 0-RTT data to at least 1200
-bytes. Additional datagrams containing only 0-RTT packets with long headers
-could be sent from the client to the server after the Client Initial datagram,
-containing the rest of the 0-RTT data. The amount of 0-RTT protected data
-that can be sent in the first flight is limited by the initial congestion
+When a 0-RTT packet is coalesced with an Initial packet, the datagram
+will be padded to 1200 byes. Additional datagrams containing only 0-RTT
```

Comment:
Note: This is also "lies to children".  If a Handshake packet is coalesced with these (which can happen, depending on how the client works; ours will do this in some circumstances), then the datagram doesn't need to be padded.

Reply:
It doesn't?  I'm missing that in: https://www.rfc-editor.org/rfc/rfc9000.html#name-initial-datagram-size

I agree that in practice, if the client has Handshake keys but hasn't used them(ie: it's sending an Initial ACK before the Handshake packet), then there's no reason to pad the datagram, but I can't find text which makes an exception for that.

Reply:
You know, I don't think that this is in the spec.  It's not a condition that the spec considers.  So maybe I can take that back.

It's what our implementation does though.  There is no point padding if you are sending Handshake, because the Handshake packet will unblock the server anti-amplification limit and you know that the server can process the Handshake packet (the only reason to send Handshake in this case is to acknowledge a Handshake packet).

Some implementations won't even send Initial in this case because they drop the keys.  We have this extra wrinkle that allows us to acknowledge Initial packets as a final gasp before dropping the keys.  It's a courtesy only and not reliable, but it means that we send Initial + Handshake + 0-RTT all in the same datagram.  AND that datagram won't be padded, even though it contains an Initial.

So I'm totally OK with this text remaining.  I certainly don't want to have to explain all of that.

Reply:
The 'courtesy ACKs' are actually useful because they provide an extra RTT sample early in the connection, so thanks for your courtesy :)